pki/pki.go

```diff
@@ -0,0 +1,179 @@
+package pki
+
+import (
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/pem"
+	"errors"
+	"fmt"
+
+	"github.com/cloudflare/circl/sign"
+	"github.com/cloudflare/circl/sign/api"
+)
+
+var allSchemesByOID map[string]sign.Scheme
+var allSchemesByTLS map[uint]sign.Scheme
+
+func init() {
+	allSchemesByOID = make(map[string]sign.Scheme)
+	for _, scheme := range api.AllSchemes() {
```

Comment:
@armfazh You can't rely on the right `init()` order.

Reply:
https://medium.com/@markbates/go-init-order-dafa89fcef22

Reply:
> If a package has imports, the imported packages are initialized before initializing the package itself.

https://golang.org/ref/spec#Package_initialization